frame/support/src/dispatch.rs

```diff
@@ -1329,21 +1329,12 @@ macro_rules! decl_module {
 		{
 			fn on_runtime_upgrade() -> $return {
 				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!("on_runtime_upgrade"));
-				let result: $return = (|| { $( $impl )* })();
```

Comment:
here I just re-ordered stuff: first log that the migration is about to happen, and then do it.